### DISCUSSION TITLE
How to get WorkflowId and  ActivityId on EditActivity page

### DISCUSSION THREAD

**Discussion by hyzx86** [OrchardCMS/OrchardCore#14298] [Q&A]
First of all, why would I want to get this information

Because the current HttpRequestEvent has some encryption problems.
For example, if I export a locally developed workflow and then import it into another workflow, this Token will not be recognized

So my plan is to develop a UserTriggerdEvent

Contains the following fields:
- A collection of user ids that can trigger the workflow
- A collection of users that can trigger the workflow
- Whether to allow all users

The node way will use an Api to trigger based on WorkflowTypeId/WorkflowId and Activityd to trigger
In my understanding they should already be fully qualified to the correct `WorkflowType` or `WorkflowInstance`

So here's the question,

How do I get the Id of the node and the workflow Id of the node in an Activity editing interface
I haven't fully mastered the use of Shape yet, so I hope someone can help me see if this problem can be solved with minimal cost

![image](https://github.com/OrchardCMS/OrchardCore/assets/15613121/502b8884-3fb8-4d6d-8830-2557477253b2)

-------------------------------

**Chinese**

首先我解释下为什么要获取这些信息

由于目前的HttpRequestEvent 存在一些 加密上的问题。
比如我将本地开发好的工作流导出去后再导入到其他工作流中会无法识别这个Token

所以 我的计划是 开发一个UserTriggerdEvent

包含以下字段：

- 可触发该工作流的用户Id集合
- 可触发该工作流的用户集合
- 是否允许所有用户

该节点的触发方式将采用一个Api 根据 WorkflowTypeId/WorkflowId 和 Activityd 来触发
在我的理解上它们应该已经可以完全限定到正确的工作流类型或流程实例

那么问题来了，
我要如何在 一个 Activity 的编辑界面获取到 该节点的Id 和 该节点所在工作流的Id 
我还没有完全掌握Shape 的用法，所以希望有人可以帮忙看看这个问题是否可以使用最小代价解决这个问题

![image](https://github.com/OrchardCMS/OrchardCore/assets/15613121/502b8884-3fb8-4d6d-8830-2557477253b2)

**Comment by weirdyang** [comment]
I solved this by creating a custom task which allow users use a custom slug instead of an encrypted string

as for importing the workflows with the default http task, I use a recipe to refresh all the urls, but this might not be useful for you as it generates a new encrypted string
